plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -1748,6 +1748,58 @@ public void testUuid()
+                .addRoundTrip("uuid[]", "ARRAY[UUID '{a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11}', UUID '00000000-0000-0000-0000-000000000000']", new ArrayType(UUID), "ARRAY[UUID 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11', UUID '00000000-0000-0000-0000-000000000000']")
+                .addRoundTrip("uuid[]", "ARRAY[UUID '{a0eebc99-9c0b4ef8-bb6d6bb9-bd380a11}', UUID '00000000-0000-0000-0000-000000000000']", new ArrayType(UUID), "ARRAY[UUID 'a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11', UUID '00000000-0000-0000-0000-000000000000']")
+
+                .addRoundTrip("uuid[]", "ARRAY[UUID '00000000-0000-0000-0000-000000000000', UUID '12151fd2-7586-11e9-8f9e-2a86e4085a59']", new ArrayType(UUID), "ARRAY[UUID '00000000-0000-0000-0000-000000000000', UUID '12151fd2-7586-11e9-8f9e-2a86e4085a59']")
+                .addRoundTrip("uuid[]", "ARRAY[UUID '123e4567-e89b-12d3-a456-426655440000', UUID 'c9c0b3f4-3e7e-4e1a-83eb-bb3e1a2c3c7e']", new ArrayType(UUID), "ARRAY[UUID '123e4567-e89b-12d3-a456-426655440000', UUID 'c9c0b3f4-3e7e-4e1a-83eb-bb3e1a2c3c7e']")
+                .addRoundTrip("uuid[]", "ARRAY[UUID 'f47ac10b-58cc-4372-a567-0e02b2c3d479', UUID '3c6f4b8e-7a3e-4d7c-bc8e-3a0fbc1c6ad1']", new ArrayType(UUID), "ARRAY[UUID 'f47ac10b-58cc-4372-a567-0e02b2c3d479', UUID '3c6f4b8e-7a3e-4d7c-bc8e-3a0fbc1c6ad1']")
+                .execute(getQueryRunner(), session, postgresCreateAndInsert("postgresql_test_array_uuid_or_empty_or_nulls"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("ARRAY(uuid)", "NULL", new ArrayType(UUID), "CAST(NULL AS ARRAY(UUID))")
+                .addRoundTrip("ARRAY(uuid)", "ARRAY[]", new ArrayType(UUID), "CAST(ARRAY[] AS ARRAY(UUID))")
+
+                .addRoundTrip("ARRAY(uuid)", "ARRAY[UUID '00000000-0000-0000-0000-000000000000', UUID '12151fd2-7586-11e9-8f9e-2a86e4085a59']", new ArrayType(UUID), "ARRAY[UUID '00000000-0000-0000-0000-000000000000', UUID '12151fd2-7586-11e9-8f9e-2a86e4085a59']")
+                .addRoundTrip("ARRAY(uuid)", "ARRAY[UUID '123e4567-e89b-12d3-a456-426655440000', UUID 'c9c0b3f4-3e7e-4e1a-83eb-bb3e1a2c3c7e']", new ArrayType(UUID), "ARRAY[UUID '123e4567-e89b-12d3-a456-426655440000', UUID 'c9c0b3f4-3e7e-4e1a-83eb-bb3e1a2c3c7e']")
+                .addRoundTrip("ARRAY(uuid)", "ARRAY[UUID 'f47ac10b-58cc-4372-a567-0e02b2c3d479', UUID '3c6f4b8e-7a3e-4d7c-bc8e-3a0fbc1c6ad1']", new ArrayType(UUID), "ARRAY[UUID 'f47ac10b-58cc-4372-a567-0e02b2c3d479', UUID '3c6f4b8e-7a3e-4d7c-bc8e-3a0fbc1c6ad1']")
+                .execute(getQueryRunner(), session, trinoCreateAsSelect(session, "trino_test_array_uuid_or_empty_or_nulls"))
+                .execute(getQueryRunner(), session, trinoCreateAndInsert(session, "trino_test_array_uuid_or_empty_or_nulls"));
+    }
+
+    @Test
+    public void testArrayNulls()
+    {
+        Session session = sessionWithArrayAsArray();
+
+        // Verify only SELECT instead of using SqlDataTypeTest because array comparison not supported for arrays with null elements
```

Comment:
Mention https://github.com/trinodb/trino/issues/11397, so we can revisit the test once we support array comparsion with null elements